### PR TITLE
Feature/145 mobile sidebar

### PR DESCRIPTION
**Pull request by RohitGarudadri**
### Tickets:

145

### List of changes:

Fixed mobile sidebar taking over the whole screen.
Fixed sidebar not scrolling with the user.
Sidebar at minimum 50px width for small screens.

### Type of change:

Please delete options that aren't relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

### How did you do this?

Changed the max width that the sidebar can take of the screen

### Why are you choosing this approach?

Only way?

### PR Checklist:

- [x] Merged `develop` branch (before testing)
- [x] Linted my code locally
- [x] Listed change(s) in the Changelog
- [x] Tested all links in project relevant browsers
- [x] Tested all links on different screen sizes
- [x] Referenced all useful info (issues, tasks, etc)

### CODE REVIEW

**Comment by pierreTklein** [comment]
Can we get a gif?

**Comment by pierreTklein** [comment]
# Bugs

### Scrollbars appear on nav

Screen Shot 2019-12-14 at 1 43 00 PM

### Mobile sidebar does not appear to work (using chrome)

Mobile Sidebar not working gif

### Application crashes when user is confirmed. 
Stack trace:
```
TypeError: Cannot read property 'trim' of undefined
Autosuggest.getQuery
node_modules/react-autosuggest/dist/Autosuggest.js:331
  328 |     var inputProps = this.props.inputProps;
  329 |     var value = inputProps.value;
  330 |     var valueBeforeUpDown = this.state.valueBeforeUpDown;
> 331 |     return (valueBeforeUpDown === null ? value : valueBeforeUpDown).trim();
      | ^  332 |   }
  333 | }, {
  334 |   key: 'render',
View compiled
Autosuggest.render
node_modules/react-autosuggest/dist/Autosuggest.js:560
  557 | });
  558 | 
  559 | var renderSuggestionData = {
> 560 |   query: this.getQuery()
      | ^  561 | };
  562 | return _react2.default.createElement(_reactAutowhatever2.default, {
  563 |   multiSection: multiSection,
View compiled
finishClassComponent
node_modules/react-dom/cjs/react-dom.development.js:15141
  15138 | } else {
  15139 |   {
  15140 |     setCurrentPhase('render');
> 15141 |     nextChildren = instance.render();
        | ^  15142 | 
  15143 |     if (debugRenderPhaseSideEffects || debugRenderPhaseSideEffectsForStrictMode && workInProgress.mode & StrictMode) {
  15144 |       instance.render();
View compiled
updateClassComponent
node_modules/react-dom/cjs/react-dom.development.js:15096
  15093 |   shouldUpdate = updateClassInstance(current$$1, workInProgress, Component, nextProps, renderExpirationTime);
  15094 | }
  15095 | 
> 15096 | var nextUnitOfWork = finishClassComponent(current$$1, workInProgress, Component, shouldUpdate, hasContext, renderExpirationTime);
        | ^  15097 | {
  15098 |   var inst = workInProgress.stateNode;
  15099 | 
View compiled
beginWork
node_modules/react-dom/cjs/react-dom.development.js:15980
  15977 | 
  15978 |     var _resolvedProps = workInProgress.elementType === _Component2 ? _unresolvedProps : resolveDefaultProps(_Component2, _unresolvedProps);
  15979 | 
> 15980 |     return updateClassComponent(current$$1, workInProgress, _Component2, _resolvedProps, renderExpirationTime);
        | ^  15981 |   }
  15982 | 
  15983 | case HostRoot:
View compiled
performUnitOfWork
node_modules/react-dom/cjs/react-dom.development.js:19102
  19099 |   startProfilerTimer(workInProgress);
  19100 | }
  19101 | 
> 19102 | next = beginWork(current$$1, workInProgress, nextRenderExpirationTime);
        | ^  19103 | workInProgress.memoizedProps = workInProgress.pendingProps;
  19104 | 
  19105 | if (workInProgress.mode & ProfileMode) {
View compiled
workLoop
node_modules/react-dom/cjs/react-dom.development.js:19143
  19140 | if (!isYieldy) {
  19141 |   // Flush work without yielding
  19142 |   while (nextUnitOfWork !== null) {
> 19143 |     nextUnitOfWork = performUnitOfWork(nextUnitOfWork);
        | ^  19144 |   }
  19145 | } else {
  19146 |   // Flush asynchronous work until there's a higher priority event
View compiled
HTMLUnknownElement.callCallback
node_modules/react-dom/cjs/react-dom.development.js:147
  144 |     window.event = windowEvent;
  145 |   }
  146 | 
> 147 |   func.apply(context, funcArgs);
      | ^  148 |   didError = false;
  149 | } // Create a global error event handler. We use this to capture the value
  150 | // that was thrown. It's possible that this error handler will fire more
View compiled
invokeGuardedCallbackDev
node_modules/react-dom/cjs/react-dom.development.js:196
  193 | // errors, it will trigger our global error handler.
  194 | 
  195 | evt.initEvent(evtType, false, false);
> 196 | fakeNode.dispatchEvent(evt);
      | ^  197 | 
  198 | if (windowEventDescriptor) {
  199 |   Object.defineProperty(window, 'event', windowEventDescriptor);
View compiled
invokeGuardedCallback
node_modules/react-dom/cjs/react-dom.development.js:250
  247 | function invokeGuardedCallback(name, func, context, a, b, c, d, e, f) {
  248 |   hasError = false;
  249 |   caughtError = null;
> 250 |   invokeGuardedCallbackImpl$1.apply(reporter, arguments);
      | ^  251 | }
  252 | /**
  253 |  * Same as invokeGuardedCallback, but instead of returning an error, it stores
View compiled
replayUnitOfWork
node_modules/react-dom/cjs/react-dom.development.js:18350
  18347 | 
  18348 | isReplayingFailedUnitOfWork = true;
  18349 | originalReplayError = thrownValue;
> 18350 | invokeGuardedCallback(null, workLoop, null, isYieldy);
        | ^  18351 | isReplayingFailedUnitOfWork = false;
  18352 | originalReplayError = null;
  18353 | 
View compiled
renderRoot
node_modules/react-dom/cjs/react-dom.development.js:19261
  19258 | if (true && replayFailedUnitOfWorkWithInvokeGuardedCallback) {
  19259 |   if (mayReplay) {
  19260 |     var failedUnitOfWork = nextUnitOfWork;
> 19261 |     replayUnitOfWork(failedUnitOfWork, thrownValue, isYieldy);
        | ^  19262 |   }
  19263 | } // TODO: we already know this isn't true in some cases.
  19264 | // At least this shows a nicer error message until we figure out the cause.
View compiled
performWorkOnRoot
node_modules/react-dom/cjs/react-dom.development.js:20165
  20162 |   cancelTimeout(timeoutHandle);
  20163 | }
  20164 | 
> 20165 | renderRoot(root, isYieldy);
        | ^  20166 | finishedWork = root.finishedWork;
  20167 | 
  20168 | if (finishedWork !== null) {
View compiled
performWork
node_modules/react-dom/cjs/react-dom.development.js:20075
  20072 |   }
  20073 | } else {
  20074 |   while (nextFlushedRoot !== null && nextFlushedExpirationTime !== NoWork && minExpirationTime <= nextFlushedExpirationTime) {
> 20075 |     performWorkOnRoot(nextFlushedRoot, nextFlushedExpirationTime, false);
        | ^  20076 |     findHighestPriorityRoot();
  20077 |   }
  20078 | } // We're done flushing work. Either we ran out of time in this callback,
View compiled
performSyncWork
node_modules/react-dom/cjs/react-dom.development.js:20049
  20046 | }
  20047 | 
  20048 | function performSyncWork() {
> 20049 |   performWork(Sync, false);
        | ^  20050 | }
  20051 | 
  20052 | function performWork(minExpirationTime, isYieldy) {
View compiled
requestWork
node_modules/react-dom/cjs/react-dom.development.js:19904
  19901 | 
  19902 | 
  19903 | if (expirationTime === Sync) {
> 19904 |   performSyncWork();
        | ^  19905 | } else {
  19906 |   scheduleCallbackWithExpirationTime(root, expirationTime);
  19907 | }
View compiled
scheduleWork
node_modules/react-dom/cjs/react-dom.development.js:19711
  19708 | !isWorking || isCommitting$1 || // ...unless this is a different root than the one we're rendering.
  19709 | nextRoot !== root) {
  19710 |   var rootExpirationTime = root.expirationTime;
> 19711 |   requestWork(root, rootExpirationTime);
        | ^  19712 | }
  19713 | 
  19714 | if (nestedUpdateCount > NESTED_UPDATE_LIMIT) {
View compiled
enqueueSetState
node_modules/react-dom/cjs/react-dom.development.js:12936
  12933 | 
  12934 |   flushPassiveEffects();
  12935 |   enqueueUpdate(fiber, update);
> 12936 |   scheduleWork(fiber, expirationTime);
        | ^  12937 | },
  12938 | enqueueReplaceState: function enqueueReplaceState(inst, payload, callback) {
  12939 |   var fiber = get(inst);
View compiled
ManageApplicationContainer.push../node_modules/react/cjs/react.development.js.Component.setState
node_modules/react/cjs/react.development.js:356
  353 | 
  354 | Component.prototype.setState = function (partialState, callback) {
  355 |   !(typeof partialState === 'object' || typeof partialState === 'function' || partialState == null) ? invariant(false, 'setState(...): takes an object of state variables to update or a function which returns an object of state variables.') : void 0;
> 356 |   this.updater.enqueueSetState(this, partialState, callback, 'setState');
      | ^  357 | };
  358 | /**
  359 |  * Forces an update. This should only be invoked when it is known with
View compiled
ManageApplicationContainer._callee$
src/features/Application/ManageApplicationContainer.tsx:134
  131 | try {
  132 |   const response = await Hacker.getSelf();
  133 |   const hackerDetails = response.data.data;
> 134 |   this.setState({
      | ^  135 |     hackerDetails,
  136 |   });
  137 | } catch (e) {
```

**Comment by RohitGarudadri** [comment]
> # Bugs
> ### Scrollbars appear on nav
> Screen Shot 2019-12-14 at 1 43 00 PM
> 
> ### Mobile sidebar does not appear to work (using chrome)
> Mobile Sidebar not working gif
> 
> ### Application crashes when user is confirmed.
> Stack trace:
> 
> ```
> TypeError: Cannot read property 'trim' of undefined
> Autosuggest.getQuery
> node_modules/react-autosuggest/dist/Autosuggest.js:331
>   328 |     var inputProps = this.props.inputProps;
>   329 |     var value = inputProps.value;
>   330 |     var valueBeforeUpDown = this.state.valueBeforeUpDown;
> > 331 |     return (valueBeforeUpDown === null ? value : valueBeforeUpDown).trim();
>       | ^  332 |   }
>   333 | }, {
>   334 |   key: 'render',
> View compiled
> Autosuggest.render
> node_modules/react-autosuggest/dist/Autosuggest.js:560
>   557 | });
>   558 | 
>   559 | var renderSuggestionData = {
> > 560 |   query: this.getQuery()
>       | ^  561 | };
>   562 | return _react2.default.createElement(_reactAutowhatever2.default, {
>   563 |   multiSection: multiSection,
> View compiled
> finishClassComponent
> node_modules/react-dom/cjs/react-dom.development.js:15141
>   15138 | } else {
>   15139 |   {
>   15140 |     setCurrentPhase('render');
> > 15141 |     nextChildren = instance.render();
>         | ^  15142 | 
>   15143 |     if (debugRenderPhaseSideEffects || debugRenderPhaseSideEffectsForStrictMode && workInProgress.mode & StrictMode) {
>   15144 |       instance.render();
> View compiled
> updateClassComponent
> node_modules/react-dom/cjs/react-dom.development.js:15096
>   15093 |   shouldUpdate = updateClassInstance(current$$1, workInProgress, Component, nextProps, renderExpirationTime);
>   15094 | }
>   15095 | 
> > 15096 | var nextUnitOfWork = finishClassComponent(current$$1, workInProgress, Component, shouldUpdate, hasContext, renderExpirationTime);
>         | ^  15097 | {
>   15098 |   var inst = workInProgress.stateNode;
>   15099 | 
> View compiled
> beginWork
> node_modules/react-dom/cjs/react-dom.development.js:15980
>   15977 | 
>   15978 |     var _resolvedProps = workInProgress.elementType === _Component2 ? _unresolvedProps : resolveDefaultProps(_Component2, _unresolvedProps);
>   15979 | 
> > 15980 |     return updateClassComponent(current$$1, workInProgress, _Component2, _resolvedProps, renderExpirationTime);
>         | ^  15981 |   }
>   15982 | 
>   15983 | case HostRoot:
> View compiled
> performUnitOfWork
> node_modules/react-dom/cjs/react-dom.development.js:19102
>   19099 |   startProfilerTimer(workInProgress);
>   19100 | }
>   19101 | 
> > 19102 | next = beginWork(current$$1, workInProgress, nextRenderExpirationTime);
>         | ^  19103 | workInProgress.memoizedProps = workInProgress.pendingProps;
>   19104 | 
>   19105 | if (workInProgress.mode & ProfileMode) {
> View compiled
> workLoop
> node_modules/react-dom/cjs/react-dom.development.js:19143
>   19140 | if (!isYieldy) {
>   19141 |   // Flush work without yielding
>   19142 |   while (nextUnitOfWork !== null) {
> > 19143 |     nextUnitOfWork = performUnitOfWork(nextUnitOfWork);
>         | ^  19144 |   }
>   19145 | } else {
>   19146 |   // Flush asynchronous work until there's a higher priority event
> View compiled
> HTMLUnknownElement.callCallback
> node_modules/react-dom/cjs/react-dom.development.js:147
>   144 |     window.event = windowEvent;
>   145 |   }
>   146 | 
> > 147 |   func.apply(context, funcArgs);
>       | ^  148 |   didError = false;
>   149 | } // Create a global error event handler. We use this to capture the value
>   150 | // that was thrown. It's possible that this error handler will fire more
> View compiled
> invokeGuardedCallbackDev
> node_modules/react-dom/cjs/react-dom.development.js:196
>   193 | // errors, it will trigger our global error handler.
>   194 | 
>   195 | evt.initEvent(evtType, false, false);
> > 196 | fakeNode.dispatchEvent(evt);
>       | ^  197 | 
>   198 | if (windowEventDescriptor) {
>   199 |   Object.defineProperty(window, 'event', windowEventDescriptor);
> View compiled
> invokeGuardedCallback
> node_modules/react-dom/cjs/react-dom.development.js:250
>   247 | function invokeGuardedCallback(name, func, context, a, b, c, d, e, f) {
>   248 |   hasError = false;
>   249 |   caughtError = null;
> > 250 |   invokeGuardedCallbackImpl$1.apply(reporter, arguments);
>       | ^  251 | }
>   252 | /**
>   253 |  * Same as invokeGuardedCallback, but instead of returning an error, it stores
> View compiled
> replayUnitOfWork
> node_modules/react-dom/cjs/react-dom.development.js:18350
>   18347 | 
>   18348 | isReplayingFailedUnitOfWork = true;
>   18349 | originalReplayError = thrownValue;
> > 18350 | invokeGuardedCallback(null, workLoop, null, isYieldy);
>         | ^  18351 | isReplayingFailedUnitOfWork = false;
>   18352 | originalReplayError = null;
>   18353 | 
> View compiled
> renderRoot
> node_modules/react-dom/cjs/react-dom.development.js:19261
>   19258 | if (true && replayFailedUnitOfWorkWithInvokeGuardedCallback) {
>   19259 |   if (mayReplay) {
>   19260 |     var failedUnitOfWork = nextUnitOfWork;
> > 19261 |     replayUnitOfWork(failedUnitOfWork, thrownValue, isYieldy);
>         | ^  19262 |   }
>   19263 | } // TODO: we already know this isn't true in some cases.
>   19264 | // At least this shows a nicer error message until we figure out the cause.
> View compiled
> performWorkOnRoot
> node_modules/react-dom/cjs/react-dom.development.js:20165
>   20162 |   cancelTimeout(timeoutHandle);
>   20163 | }
>   20164 | 
> > 20165 | renderRoot(root, isYieldy);
>         | ^  20166 | finishedWork = root.finishedWork;
>   20167 | 
>   20168 | if (finishedWork !== null) {
> View compiled
> performWork
> node_modules/react-dom/cjs/react-dom.development.js:20075
>   20072 |   }
>   20073 | } else {
>   20074 |   while (nextFlushedRoot !== null && nextFlushedExpirationTime !== NoWork && minExpirationTime <= nextFlushedExpirationTime) {
> > 20075 |     performWorkOnRoot(nextFlushedRoot, nextFlushedExpirationTime, false);
>         | ^  20076 |     findHighestPriorityRoot();
>   20077 |   }
>   20078 | } // We're done flushing work. Either we ran out of time in this callback,
> View compiled
> performSyncWork
> node_modules/react-dom/cjs/react-dom.development.js:20049
>   20046 | }
>   20047 | 
>   20048 | function performSyncWork() {
> > 20049 |   performWork(Sync, false);
>         | ^  20050 | }
>   20051 | 
>   20052 | function performWork(minExpirationTime, isYieldy) {
> View compiled
> requestWork
> node_modules/react-dom/cjs/react-dom.development.js:19904
>   19901 | 
>   19902 | 
>   19903 | if (expirationTime === Sync) {
> > 19904 |   performSyncWork();
>         | ^  19905 | } else {
>   19906 |   scheduleCallbackWithExpirationTime(root, expirationTime);
>   19907 | }
> View compiled
> scheduleWork
> node_modules/react-dom/cjs/react-dom.development.js:19711
>   19708 | !isWorking || isCommitting$1 || // ...unless this is a different root than the one we're rendering.
>   19709 | nextRoot !== root) {
>   19710 |   var rootExpirationTime = root.expirationTime;
> > 19711 |   requestWork(root, rootExpirationTime);
>         | ^  19712 | }
>   19713 | 
>   19714 | if (nestedUpdateCount > NESTED_UPDATE_LIMIT) {
> View compiled
> enqueueSetState
> node_modules/react-dom/cjs/react-dom.development.js:12936
>   12933 | 
>   12934 |   flushPassiveEffects();
>   12935 |   enqueueUpdate(fiber, update);
> > 12936 |   scheduleWork(fiber, expirationTime);
>         | ^  12937 | },
>   12938 | enqueueReplaceState: function enqueueReplaceState(inst, payload, callback) {
>   12939 |   var fiber = get(inst);
> View compiled
> ManageApplicationContainer.push../node_modules/react/cjs/react.development.js.Component.setState
> node_modules/react/cjs/react.development.js:356
>   353 | 
>   354 | Component.prototype.setState = function (partialState, callback) {
>   355 |   !(typeof partialState === 'object' || typeof partialState === 'function' || partialState == null) ? invariant(false, 'setState(...): takes an object of state variables to update or a function which returns an object of state variables.') : void 0;
> > 356 |   this.updater.enqueueSetState(this, partialState, callback, 'setState');
>       | ^  357 | };
>   358 | /**
>   359 |  * Forces an update. This should only be invoked when it is known with
> View compiled
> ManageApplicationContainer._callee$
> src/features/Application/ManageApplicationContainer.tsx:134
>   131 | try {
>   132 |   const response = await Hacker.getSelf();
>   133 |   const hackerDetails = response.data.data;
> > 134 |   this.setState({
>       | ^  135 |     hackerDetails,
>   136 |   });
>   137 | } catch (e) {
> ```



> Hey! Check out the items I left above. Not sure how much of these issues stem from this PR, but I am not able to see the mobile sidebar properly.

Not sure about the crashing when confirmed:/ Don't think that one is based on my changes though. I am unable to reproduce the scroll bars on the nav though

**Comment by pierreTklein** [comment]
I'm using MacOS on Chrome, if that helps at all with your debugging.

**Comment by pierreTklein** [comment]
Crash was due to issue with old data being used to test this.

**Comment by pierreTklein** [comment]
Still getting the mobile sidebar issue:

File: src/features/Sidebar/SidebarContainer.tsx [hunk 1]
```diff
@@ -9,10 +9,11 @@ export const SidebarContainer = styled.nav`
   transform: 'translateX(-100%)';
   height: 100%;
   width: 18%;
+  min-width: 50px;
   max-width: 18%;
-  text-align: left;
   z-index: 10;
-  padding: 0 0 0 0;
+  text-align: left;
+  padding: 0 0 0 0%;
```
Comment (krubenok) [inline src/features/Sidebar/SidebarContainer.tsx]: why is there a % here?

**Pull request closed without merging**